mm2src/coins/utxo/utxo_tests.rs

```diff
@@ -4322,49 +4326,112 @@ fn test_block_header_utxo_loop() {
+        *expected_steps.lock().unwrap() = vec![(15, 18)];
+        unsafe { CURRENT_BLOCK_COUNT = 18 }
         Timer::sleep(2.).await;
-        let get_headers_count = client.block_headers_storage().get_last_block_height().await.unwrap();
-        assert_eq!(get_headers_count, 17);
+        let get_headers_count = client
+            .block_headers_storage()
+            .get_last_block_height()
+            .await
+            .unwrap()
+            .unwrap();
+        assert_eq!(get_headers_count, 18);
         assert!(expected_steps.lock().unwrap().is_empty());
 
-        *expected_steps.lock().unwrap() = vec![(18, 18)];
-        unsafe { CURRENT_BLOCK_COUNT = 18 }
+        *expected_steps.lock().unwrap() = vec![(19, 19)];
+        unsafe { CURRENT_BLOCK_COUNT = 19 }
         Timer::sleep(2.).await;
-        let get_headers_count = client.block_headers_storage().get_last_block_height().await.unwrap();
-        assert_eq!(get_headers_count, 18);
+        let get_headers_count = client
+            .block_headers_storage()
+            .get_last_block_height()
+            .await
+            .unwrap()
+            .unwrap();
+        assert_eq!(get_headers_count, 19);
         assert!(expected_steps.lock().unwrap().is_empty());
 
-        *expected_steps.lock().unwrap() = vec![(19, 22), (23, 25)];
-        unsafe { CURRENT_BLOCK_COUNT = 25 }
-        Timer::sleep(3.).await;
-        let get_headers_count = client.block_headers_storage().get_last_block_height().await.unwrap();
-        assert_eq!(get_headers_count, 25);
+        *expected_steps.lock().unwrap() = vec![(20, 23), (24, 26)];
+        unsafe { CURRENT_BLOCK_COUNT = 26 }
+        Timer::sleep(4.).await;
+        let get_headers_count = client
+            .block_headers_storage()
+            .get_last_block_height()
+            .await
+            .unwrap()
+            .unwrap();
+        assert_eq!(get_headers_count, 26);
         assert!(expected_steps.lock().unwrap().is_empty());
     };
 
     if let Either::Left(_) = block_on(futures::future::select(loop_fut.boxed(), test_fut.boxed())) {
         panic!("Loop shouldn't stop")
     };
 }
+
+#[test]
+fn test_spv_conf_with_verification() {
+    // Block header hash for BLOCK HEIGHT 2016
+    let hash = "00000000a141216a896c54f211301c436e557a8d55900637bbdce14c6c7bddef".into();
+    // test for good retarget_block_header_height
+    let mut spv_conf = SPVConf {
+        block_header: SPVBlockHeader {
+            height: 4032,
+            hash,
+            time: 1234466190,
+            bits: BlockHeaderBits::Compact(486604799.into()),
+        },
+        max_stored_block_headers: None,
+        validation_params: Some(BlockHeaderValidationParams {
+            difficulty_check: false,
+            constant_difficulty: false,
+            difficulty_algorithm: Some(DifficultyAlgorithm::BitcoinMainnet),
+        }),
+    };
+    assert!(spv_conf.validate("BTC").is_ok());
+
+    // test for bad retarget_block_header_height
```

Comment:
Can you also add a test case for bad `max_stored_block_headers`

Reply:
done